ui/src/testing/stubs/workspaces-api-stub.ts

```diff
@@ -105,31 +131,67 @@ export class WorkspacesApiStub extends WorkspacesApi {
     });
   }
 
+  private duplicateWorkspaceImpl(
```

Comment:
extracted from the `cloneWorkspace()` stub so it can also be used by duplicateWorkspaceAsync()